### DISCUSSION TITLE
Importing SDL2 functions in an extern file

### DISCUSSION THREAD

**Discussion by AlectronikForge** [edubart/nelua-lang#38] [Q&A] [answered]
I tried to split the SDL example into two files, as a base to create a SDL binding. So I moved all the SDL related stuff into sdl.nelua, deleted all the local's and added a require 'sdl' statement.

But then the compiler complains about the symbols in sdl.nelua being undeclared.

main.nelua:
`require 'sdl'
SDL_Init(SDL_INIT_VIDEO)`

`main.nelua:2:9: from: AST node Call
require 'sdl'
        ^~~~~
/run/media/nix/BuildPart/sdl.nelua:45:2: from: AST node FuncDef
 function SDL_Init(flags: uint32): int32 <cimport> end
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/media/nix/BuildPart/sdl.nelua:45:11: error: undeclared symbol 'SDL_Init'
 function SDL_Init(flags: uint32): int32 <cimport> end
          ^~~~~~~~`

Or if I leave the 'local' there, it (as intended) doesn't find SDL_Init at all. Global doesn't help either, I have to declare an empty record and put the functions in there:

`global Sdl = @record{}
function Sdl.Init(flags: uint32): int32 <cimport 'SDL_Init'> end`

Is this intended behaviour?

(Sorry, I can't get the line breaks right)

**Comment by edubart** [comment, marked as answer]
For complete SDL2 bindings I suggest to use https://github.com/edubart/nelua-decl/blob/main/libs/sdl2/sdl2.nelua
Check in this file how I export symbols to other files, also see the sdl-test.nelua in the same repository as a usage example.

**Comment by AlectronikForge** [comment]
Thanks. Interesting, I must have made some mistake because I couldn't get it work without a record (which is comfortable anyways) but I'll have a deeper look and contribute some bindings if I find the time to make them :)

Is there a forum or chat about nelua yet? I really like the language and would like to use it later to write an OS kernel for which it appears to be well suited to me, as I'd just have to implement some basic libc functions to get it running but would be nice to talk with other nelua people!

**Comment by Andre-LA** [comment]
> Is there a forum or chat about nelua yet?

There is the Discord chat server:
https://discord.gg/7aaGeG7
